### PR TITLE
Package.suggestedTotal

### LINKED ISSUE #684: PROLONG Page should display previous price
- [ ] Prolong userPrice=true, change copy of going back to regular price
Screenshot 2021-11-24 at 15 09 59

### PR DESCRIPTION
**Pull request by pae-blue**
`Package.suggestedTotal` may return a price suggestion for a custom package. It springs into action when a user's pledge total deviated from default price a membership period before.

### CODE REVIEW

**Comment by patrickvenetz** [comment]
Meanwhile, to test sufficently:
- [ ] AutoPay flow, no changes
- [ ] 1st time PROLONG, no changes
- [ ] 2nd time PROLONG, no changes
- [ ] previous pledge with goodies and donation (no `suggestdTotal`)
- [ ] previous pledge with donation (returns `suggestedTotal`)
- [ ] previous reduces pledge (returns `suggestedTotal`)
- [ ] pledge with one goodie (no change)
- [ ] pledge with 2 goodies (no change)

File: packages/republik-crowdfundings/lib/CustomPackages/index.js [hunk 1]
```diff
@@ -267,112 +292,121 @@ const getCustomOptions = async (package_) => {
...
+      ),
+    )
+    // Sort by userID, own ones up top.
+    .sort((a, b) =>
+      descending(
+        a.membership && a.membership.userId === package_.user.id,
+        b.membership && b.membership.userId === package_.user.id,
+      ),
+    )
+    // Sort by sortOrder at lat
```
Comment (fuenkchen) [inline packages/republik-crowdfundings/lib/CustomPackages/index.js:387]:
```suggestion
    // Sort by sortOrder at last
```

File: packages/republik-crowdfundings/lib/CustomPackages/index.js [hunk 2]
```diff
@@ -267,112 +292,121 @@ const getCustomOptions = async (package_) => {
...
+    .sort((a, b) => ascending(a.order, b.order))
+
+  if (!filteredAndSortedOptions.length) {
+    return
+  }
+
+  const suggestedTotal = filteredAndSortedOptions
+    .map((po) => po.suggestedPrice)
```
Comment (fuenkchen) [inline packages/republik-crowdfundings/lib/CustomPackages/index.js:395]:
```suggestion
    .map((option) => option.suggestedPrice)
```

**Review by fuenkchen — APPROVED** [review]
Only detail suggestions, maybe also add an explanatory comment above evaluate method (if possible), but in general 👍

**Comment by pae-blue** [comment]
Fixes #684

**Pull request merged**